File: frontend/src/app/features/alerts/models/alert-response.ts
import { BaseModel } from "../../../core/models/base-model";
import { AlertSeverity } from "./alert-severity-enum";
import { AlertStatus } from "./alert-status-enum";

export interface AlertResponse extends BaseModel {
  appId?: string;
  message?: string;
  severity?: AlertSeverity;
  occurrence: number;
  status: AlertStatus;
  occurredAt: string;
}

File: http-client/MCS.WatchTower.WebApi.DataTransferObjects/Responses/AlertResponse.cs
using MCS.WatchTower.WebApi.DataTransferObjects.Enumerations;

namespace MCS.WatchTower.WebApi.DataTransferObjects.Responses;

public record AlertResponse : IBaseDto
{
    public string Id { get; init; }
    public string AppId { get; init; }
    public string Message { get; init; }
    public AlertSeverity Severity { get; init; }
    public int Occurrence { get; init; }
    public AlertStatus Status { get; init; }
    public DateTime OccurredAt { get; init; }
}
